service/history/task/transfer_active_task_executor.go

```diff
@@ -1134,7 +1153,12 @@ func (t *transferActiveTaskExecutor) signalExternalExecutionFailed(
 	err := t.updateWorkflowExecution(ctx, wfContext, true,
 		func(ctx context.Context, mutableState execution.MutableState) error {
 			if !mutableState.IsWorkflowExecutionRunning() {
-				return &types.EntityNotExistsError{Message: "Workflow is not running."}
+				return execution.VersionBasedError(
+					&types.WorkflowExecutionAlreadyCompletedError{Message: "Workflow execution already completed."},
+					"1.7.0",
+					yarpc.CallFromContext(ctx).Header(common.FeatureVersionHeaderName),
+					&types.EntityNotExistsError{Message: "Workflow execution already completed."},
+				)
```

Comment:
My understanding it that all calls from history service will not go through client, so there's no version here. Same for other checks in the file.